app/code/Magento/Customer/Test/Unit/Model/ForgotPasswordToken/GetCustomerByTokenTest.php

```diff
@@ -0,0 +1,145 @@
+
+    /**
+     * @var CustomerRepositoryInterface|MockObject
+     */
+    protected $customerRepositoryMock;
+
+    /**
+     * @var CustomerSearchResultsInterface|MockObject
+     */
+    protected $searchResultMock;
+
+    /**
+     * @var CustomerInterface|MockObject
+     */
+    protected $customerMock;
+
+    /**
+     * @var GetCustomerByToken;
+     */
+    protected $model;
+
+    /**
+     * @inheritDoc
+     */
+    public function setUp()
+    {
+        $this->searchCriteriaBuilderMock = $this->createMock(SearchCriteriaBuilder::class);
+        $this->searchCriteriaMock = $this->createMock(SearchCriteria::class);
+        $this->searchResultMock = $this->createMock(CustomerSearchResultsInterface::class);
+        $this->customerRepositoryMock = $this->createMock(CustomerRepositoryInterface::class);
+        $this->customerMock = $this->getMockForAbstractClass(CustomerInterface::class);
+
+        $objectManagerHelper = new ObjectManagerHelper($this);
+        $this->model = $objectManagerHelper->getObject(
+            GetCustomerByToken::class,
+            [
+                'searchCriteriaBuilder' => $this->searchCriteriaBuilderMock,
+                'customerRepository' => $this->customerRepositoryMock
+            ]
+        );
+
+        $this->searchCriteriaBuilderMock->expects($this->once())
+            ->method('create')
+            ->willReturn($this->searchCriteriaMock);
+        $this->customerRepositoryMock->expects($this->once())
+            ->method('getList')
+            ->with($this->searchCriteriaMock)
+            ->willReturn($this->searchResultMock);
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecute(): void
+    {
+        $totalCount = 1;
+        $this->searchResultMock->expects($this->any())
+            ->method('getTotalCount')
+            ->willReturn($totalCount);
+        $this->searchResultMock->expects($this->once())
+            ->method('getItems')
+            ->willReturn([$this->customerMock]);
+
+        $this->assertInstanceOf(
+            CustomerInterface::class,
+            $this->model->execute(self::RESET_PASSWORD)
+        );
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecuteWithNoSuchEntityException(): void
+    {
+        $totalCount = 0;
+        $this->searchResultMock->expects($this->any())
+            ->method('getTotalCount')
+            ->willReturn($totalCount);
+        $this->expectExceptionObject(new NoSuchEntityException(
+            new Phrase(
+                'No such entity with rp_token = %value',
+                ['value' => self::RESET_PASSWORD]
+            )
+        ));
+
+        $this->model->execute(self::RESET_PASSWORD);
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecuteWithExpireException(): void
+    {
+        $totalCount = 2;
+        $this->searchResultMock->expects($this->any())
+            ->method('getTotalCount')
+            ->willReturn($totalCount);
```

Comment:
```suggestion
        $this->searchResultMock->method('getTotalCount')->willReturn($totalCount);
```
If it does not matter how many the method is being called, this means it's stub, not mock. Then you should avoid ->expects() at all.